tests/functional/conftest.py

```diff
@@ -167,31 +215,41 @@ def _wait(timeout=30, step=0.5):
 
 
 @pytest.fixture(scope="session")
-def gitlab_config(check_is_alive, docker_ip, docker_services, temp_dir, fixture_dir):
+def gitlab_config(
+    check_is_alive,
+    gitlab_container_name: str,
+    gitlab_url: str,
+    docker_services,
+    temp_dir: pathlib.Path,
+    fixture_dir: pathlib.Path,
+):
     config_file = temp_dir / "python-gitlab.cfg"
-    port = docker_services.port_for("gitlab", 80)
 
     start_time = time.perf_counter()
     logging.info("Waiting for GitLab container to become ready.")
     docker_services.wait_until_responsive(
-        timeout=300,
+        timeout=400,
```

Comment:
From what I could tell in the logs, the issue was happening way before the timeout actually (e.g. already at 2.5 minutes or so), when the logs state reconfgiured. Maybe we could keep it at 300 and see how that goes? 300 seconds is already a sad amount of time for a service to start (:grin:) and maybe these new checks can improve our handling of startup.

Reply:
Sure. Done. But I guess I don't see the issue with making it even 1,000 seconds 😆 Since if it doesn't come up in the timeout period then the job fails.

Reply:
Hehe I think I (at least used to) run these tests locally rather than in CI more often than you - basically I don't want people to get stuck in the dark (with an unhealthy container that's never getting out of 502 for whatever reason) for too long when running tests. So at some point we have to say "ok, something is really wrong, this container will never properly start up"